CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/MainActivity.java

```diff
@@ -3,35 +3,40 @@
 
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
-import android.util.Log;
 import android.widget.TextView;
 
 import java.io.IOException;
-import java.net.Inet4Address;
-import java.net.InetAddress;
-import java.net.NetworkInterface;
-import java.net.SocketException;
-import java.util.Enumeration;
 
-import com.couchbase.CouchbaseLiteServ.server.Server;
+import com.couchbase.javacommon.Context;
+import com.couchbase.javalistener.Server;
+import com.couchbase.javacommon.log.Log;
 
 
-public class MainActivity extends AppCompatActivity {
-    public static final String ip = getLocalIpAddress();
-    private static final String TAG = "MAIN";
 
+public class MainActivity extends AppCompatActivity{
+    private static final String TAG = "MAIN";
     private Server server;
     private final static int PORT = 8080;
-
+    private static Context context;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        context = new AndroidContext();
+        String ip = context.getLocalIpAddress();
+
         setContentView(R.layout.activity_main);
         TextView textView = findViewById(R.id.textView);
-        textView.setText(getString(R.string.server_running, MainActivity.ip, PORT));
+        textView.setText(getString(R.string.server_running, ip, PORT));
+        Log.init(new AndroidLogger());
+
         try {
-            server = new Server(MainActivity.ip, PORT);
+            Server.context = context;
```

Comment:
Perhaps nit-picky, in this application.  It is Java SOP that fields are either final or private.  Best practice would be to use a setter, here, like you do for the IPAddress, below.

Reply:
addressed this too